src/lex.cpp

```diff
@@ -450,10 +450,34 @@ void lexer::parse_current_token() {
         }
 
       case '\n':
-      case '\r':
+      case '\r': {
+        char8* c_up_to_first_matching_quote = nullptr;
```

Comment:
Nit: I'd name this variable `matching_quote`, and decrement it. (See my comment on lines 466-470.)